### PR TITLE
Add more vscode workspace settings and recommended extensions

### PR DESCRIPTION
**Pull request by believer-ufa**
According the boilerplate.

This settings add recommended extensions to project folder: prettier and eslint.

With properly configured prettier plugin, user can format his code according eslint rules, it's really nice feature.

This settings also includes `search.exclude` option to make search faster in VSCode.

Option `javascript.preferences.importModuleSpecifier` configures auto-import behavior to add full path to imported module (from `./src` folder, of course) instead relative path from current file.

`jsconfig.json` file also configures project to correctly open it in VSCode and get comfortable workflow with working import suggestions and experimental decorators support.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "printWidth": 90
```
Comment (tomatau) [inline .prettierrc]: i don't use prettier, don't want to force it on anyone either, it has annoying quirks about it's preferences for cramming code into the max line width.
anyone using the boilerplate can use it, i have eslint setup which is less aggressive and anyone can disable.

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+	// See http://go.microsoft.com/fwlink/?LinkId=827846 to learn about workspace recommendations.
+	// Extension identifier format: ${publisher}.${name}. Example: vscode.csharp
+
+	// List of extensions which should be recommended for users of this workspace.
+	"recommendations": [
+		"esbenp.prettier-vscode",
+		"dbaeumer.vscode-eslint"
```
Comment (tomatau) [inline .vscode/extensions.json]: again i don't use prettier

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,3 +1,17 @@
 {
+  "editor.tabSize": 2,
```
Comment (tomatau) [inline .vscode/settings.json]: this is configured using .editorconfig already

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,3 +1,17 @@
...
+  "editor.formatOnSave": false,
```
Comment (tomatau) [inline .vscode/settings.json]: this is a setting for each developer to choose, shouldn't be forced by project

File: .vscode/settings.json [hunk 5]
```diff
@@ -1,3 +1,17 @@
...
+  "prettier.eslintIntegration": true,
+  "files.associations": {
+    "*.js": "javascriptreact"
+  },
+  "search.exclude": {
+    "**/node_modules": true,
+    "src/static": true
+  },
+  "javascript.preferences.importModuleSpecifier": "non-relative",
+  "prettier.printWidth": 90,
```
Comment (tomatau) [inline .vscode/settings.json]: again, i don't use prettier

File: jsconfig.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "compilerOptions": {
+    "baseUrl": "./src",
+    "experimentalDecorators": true
```
Comment (tomatau) [inline jsconfig.json]: this is a duplicate of the setting already in `.vscode/settings.json`, i chose the file inside a folder that's explicitly named vscode for clarity. jsconfig.json is a file i really don't like if i'm honest!

**Comment by tomatau** [comment]
sorry i'm not going to merge this

File: jsconfig.json (hunk 6 above)
Comment (believer-ufa) [reply jsconfig.json]: VSCode doesn't suggest correctly files in your project without that settings )

Maybe just experimentalDecorators is option that can be also configured though `settings.json`

Comment (tomatau) [reply jsconfig.json]: Yes I was only talking about experimentalDecorators.

I have some local settings on a branch I'm working on for docker images and cross browser testing.

E.g.g

```json
{
  "compilerOptions": {
    "plugins": [
      {
        "name": "typescript-snapshots-plugin",
        "snapshotDir": "./",
        "snapshotCallIdentifiers": [
          "toMatchSnapshot",
          "toThrowErrorMatchingSnapshot",
          "matchSnapshot",
        ]
      }
    ],
  }
}
```

I'm happy to add the `baseUrl` here too, sounds useful!

**Pull request closed without merging**